.github/workflows/combine-prs.yml

```diff
@@ -52,20 +53,33 @@ jobs:
               if (pull['user']['login'].match(/^(dependabot\[bot\]|dependabot-preview\[bot\])$/)) {
                 console.log('Branch matched: ' + branch);
                 statusOK = true;
+                // if the PR is not mergeable, bail early.
+                const pull_request = await github.rest.pulls.get({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  pull_number: pull['number']
+                });
+                if(pull_request.data.mergeable != true) {
```

Comment:
Something I learned today is that the `github` client has a `rest` endpoint, and this returns objects that can be used instead of array accessors. If this works well, we could explore converting the rest of the code to use these accessors.